src/components/AzureMapFeature/useFeature.ts

```diff
@@ -1,28 +1,36 @@
 import { useEffect } from 'react'
 import { DataSourceType, IAzureMapFeature, ShapeType, FeatureType } from '../../types'
 import { useCheckRef } from '../../hooks/useCheckRef'
+import atlas from 'azure-maps-control'
 
 export const useFeature = (
   { setCoords, setProperties }: IAzureMapFeature,
   dataSourceRef: DataSourceType | null,
   shapeRef: ShapeType | null,
   featureRef: FeatureType | null
 ) => {
-  // Simple feature's usecases and methods
-  useCheckRef<DataSourceType, FeatureType>(dataSourceRef, featureRef, (dref, fref) => {
-    dref.add(fref)
-    return () => {
-      dref.remove(fref)
-    }
-  })
+  if(dataSourceRef instanceof atlas.source.DataSource){
+    // Simple feature's usecases and methods
+    useCheckRef<atlas.source.DataSource, FeatureType>(dataSourceRef, featureRef, (dref, fref) => {
+      dref.add(fref)
+      return () => {
+        dref.remove(fref)
+      }
+    })
 
-  // Shape's usecases and methods
-  useCheckRef<DataSourceType, ShapeType>(dataSourceRef, shapeRef, (dref, sref) => {
-    dref.add(sref)
-    return () => {
-      dref.remove(sref)
-    }
-  })
+    // Shape's usecases and methods
+    useCheckRef<atlas.source.DataSource, ShapeType>(dataSourceRef, shapeRef, (dref, sref) => {
+      dref.add(sref)
+      return () => {
+        dref.remove(sref)
+      }
+    })
+  } else if (dataSourceRef instanceof atlas.source.VectorTileSource) {
+    useCheckRef<FeatureType, FeatureType>(featureRef, featureRef, (fref) => 
```

Comment:
Can you give a little bit explanation of this logic? 
We have both `(featureRef, featureRef,` in the hooks call

Reply:
the logic exactly is: on featureRef changes when featureRef is non-null. The reason for this logic was for the sake of logging the error message where feature cannot be added to a tile source.

For typescript VectorTileSource doesn't allow AzureMapFeature to be present as child, otherwise in js the error message gives a valuable(I guess) hint for the user.